File: MES_SPRING/src/main/java/com/hm/mes_final_260106/dto/WorkerResDto.java
package com.hm.mes_final_260106.dto;

import com.hm.mes_final_260106.entity.Worker;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.Arrays;
import java.util.List;

@Data
@Builder
public class WorkerResDto {

    private Long workerId;
    private String name;
    private String dept;
    private String shift;
    private String status;
    private LocalDate joinDate;
    private List<String> certifications;

    public static WorkerResDto fromEntity(Worker w) {
        List<String> certList =
                (w.getCertifications() == null || w.getCertifications().isBlank())
                        ? List.of()
                        : Arrays.stream(w.getCertifications().split(","))
                        .map(String::trim)
                        .toList();

        return WorkerResDto.builder()
                .workerId(w.getId())
                .name(w.getName())
                .dept(w.getDept())
                .shift(w.getShift())
                .status(w.getStatus())
                .joinDate(w.getJoinDate())
                .certifications(certList)
                .build();
    }
}
///1111111111

File: MES_EQUIPMENT/Models/WorkerResDto.cs
using System;
using System.Collections.Generic;

public class WorkerResDto
{
    public long WorkerId { get; set; }
    public long MemberId { get; set; }
    public string Code { get; set; }
    public string Name { get; set; }
    public string Dept { get; set; }
    public string Shift { get; set; }
    public string Status { get; set; }
    public DateTime JoinDate { get; set; }
    public List<string> Certifications { get; set; }
}